File: Merge Strings Alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string result = "";
        int i = 0, j = 0;
        int n1 = word1.length(), n2 = word2.length();

        // Add characters alternately from both strings
        while (i < n1 && j < n2) {
            result += word1[i++];
            result += word2[j++];
        }

        // Append remaining characters from word1 (if any)
        while (i < n1) {
            result += word1[i++];
        }

        // Append remaining characters from word2 (if any)
        while (j < n2) {
            result += word2[j++];
        }

        return result;
    }
};

File: Merge Strings Alternately.py
class Solution:
    def mergeAlternately(self, word1, word2):
        result = ""
        n = len(word1)
        m = len(word2)

        for i in range(min(n, m)):
            result += word1[i]
            result += word2[i]

        if n > m:
            result += word1[m:]
        else:
            result += word2[n:]

        return result
